### ISSUE TITLE
Dual site issue with Alexa device discovery

### ISSUE THREAD

**Issue by prplrhd** [bwssytems/ha-bridge#1176]
Wondering if others have seen this issue before. Previously I've not had problems running 60 or more devices between two different homes. It worked like you would expect, habridge running on pi in both locations, Discovery command discovers everything at the two homes/LANs. This is on a single Amazon account with 5 Dots split between the two homes.

I started having problems months ago adding and removing devices, getting duplicates on discoveries etc. Now after I have stripped down everything, including forgetting all devices, I get a ping pong effect. That is, Alexa will discover the devices in one home, or the other. Max of about 22 devices. But not both. And what's weird, if it discovers 21 devices at home A, it will include one at home B. Or vice versa. seems completely random. Sometimes i get 27 devices, mostly only 22.

Both running pi with v5.3.0. Both work fine when Alexa discovers device. One site has the following in the log when discovery is successful:

01-27-2020 15:26:33.779 | INFO | Traceupnp: SSDP M-SEARCH packet from 192.168.2.154:50000 | com.bwssystems.HABridge.upnp.UpnpListener
-- | -- | -- | --
01-27-2020 15:26:34.430 | INFO | Traceupnp: send upnp discovery template 1 with response address: 192.168.2.198:80 to address: /192.168.2.154:50000 | com.bwssystems.HABridge.upnp.UpnpListener
01-27-2020 15:26:34.445 | INFO | Traceupnp: request of description.xml from: 192.168.2.154:80 filled in with address: 192.168.2.198:80 | com.bwssystems.HABridge.upnp.UpnpSettingsResource
01-27-2020 15:26:34.460 | INFO | Traceupnp: hue lights list requested by user: B5qPqJjfEtk7gezrm2uYOGwr6hD6BkENMwb21oLU from address: 192.168.2.154 | com.bwssystems.HABridge.hue.HueMulator
01-27-2020 15:26:35.082 | INFO | Traceupnp: send upnp discovery template 2 with response address: 192.168.2.198:80 to address: /192.168.2.154:50000 | com.bwssystems.HABridge.upnp.UpnpListener
01-27-2020 15:26:35.733 | INFO | Traceupnp: send upnp discovery template 3 with response address: 192.168.2.198:80 to address: /192.168.2.154:50000 | com.bwssystems.HABridge.upnp.UpnpListener

Where 2.154 is the Dot. Notice it's using port 50000 as default config'd in habridge and the Hue lights list response.

Other site has this in log:

01-27-2020 15:00:12.028 | INFO | Traceupnp: SSDP M-SEARCH packet from 192.168.1.186:1900 | com.bwssystems.HABridge.upnp.UpnpListener
-- | -- | -- | --
01-27-2020 15:00:12.073 | WARN | PTR record of class: CLASS_IN index 1, there was a problem reading the service name of the answer for domain: _services._dns-sd._udp.local. | javax.jmdns.impl.DNSIncoming

The WARN message repeats many times over several minutes. Note the response showing on port 1900 not 50000

Eventually the system where devices are not discovers settles to this pattern

01-27-2020 15:01:28.272 | INFO | Traceupnp: SSDP M-SEARCH packet from 192.168.1.186:1900 | com.bwssystems.HABridge.upnp.UpnpListener
-- | -- | -- | --
01-27-2020 15:01:28.923 | INFO | Traceupnp: send upnp discovery template 1 with response address: 192.168.1.206:80 to address: /192.168.1.186:1900 | com.bwssystems.HABridge.upnp.UpnpListener
01-27-2020 15:01:29.574 | INFO | Traceupnp: send upnp discovery template 2 with response address: 192.168.1.206:80 to address: /192.168.1.186:1900 | com.bwssystems.HABridge.upnp.UpnpListener
01-27-2020 15:01:30.226 | INFO | Traceupnp: send upnp discovery template 3 with response address: 192.168.1.206:80 to address: /192.168.1.186:1900 | com.bwssystems.HABridge.upnp.UpnpListener

Repeating about every 2 seconds.

And again the behavior is identical between the two pi units. Whichever one Alexa decides to listen to, it responds on port 50000 and handshakes ok, asks for the "light list". The one that doesn't work it logs as trying to communicate on 1900. And fails. 

Pi units are all apt-get update, apt-get dist-upgrade current. Nothing else going on over the network but given the behavior presents in both places, it doesn't appear to be network issue.

I have not yet tried a rollback to 4.5.6 nor running with separate Amazon accounts. One explanation is Amazon is locking down on multi site use like I've been doing for years.

**Comment by bwssytems** [comment]
Dual site with the same account could cause issues. You will be at the mercy of Amazon changing how an Alexa works. I have not seen where they resolve this as they link all Alexas on the same account.

**Comment by prplrhd** [comment]
Yes that's correct, linking on the same account up to this point has been a feature, not a bug. Until a few months ago, all devices were discovered properly and registered with Alexa, totaling over 60. Now it finds max 22-27 and typically only from one location. 

Interested in your view of the WARN messages that show up in the log on the pi running habridge where discovery fails.

020 15:00:12.073 | WARN | PTR record of class: CLASS_IN index 1, there was a problem reading the service name of the answer for domain: _services._dns-sd._udp.local. | javax.jmdns.impl.DNSIncoming
-- | -- | -- | --

This is just a sample, during discovery there are at times dozens of these entries. It often appears as if each log entry corresponds to a failed device discovery.

**Comment by bwssytems** [comment]
That is a message from the MDNS service in the ha-bridge. MDNS is used by some clients trying to find the bridge but the Amazon Echo does not use MDNS, either nupnp (proprietary to philips) or upnp which the ha-bridge will answer.
